gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/diagnosticreport/SpecimenBatteryMapperTest.java

```diff
@@ -37,144 +35,297 @@
+                <id root="ENCOUNTER_ID"/>
+                <author typeCode="AUT" contextControlCode="OP">
+                    <time value="20220302105070"/>
+                    <agentRef classCode="AGNT">
+                        <id root="749107A2-4975-441F-8EDF-ADFF451FD12D"/>
+                    </agentRef>
+                </author>
+                <component>
+                    <CompoundStatement classCode="CLUSTER">
+                        <id root="DR_TEST_ID"/>
+                        <component>
+                            <CompoundStatement classCode="CLUSTER">
+                                <id root="TEST_SPECIMEN_ID_1"/>
+                                <component typeCode="COMP" contextConductionInd="true">
+                                    <CompoundStatement classCode="BATTERY" moodCode="EVN">
+                                        <id root="SPECIMEN_CHILD_BATTERY_COMPOUND_STATEMENT_ID_1"/>
+                                    </CompoundStatement>
+                                </component>
+                            </CompoundStatement>
+                        </component>
+                    </CompoundStatement>
+                </component>
+            </ehrComposition>
+            """;
+
+        final var ehrExtract = unmarshallEhrExtractFromEhrCompositionXml(ehrCompositionXml);
+        final var batteryCompoundStatement = getBatteryCompoundStatements(ehrExtract);
+        final var batteryParameters = getSpecimenBatteryParameters(
+            ehrExtract,
+            batteryCompoundStatement,
+            getObservations(),
+            getObservationComments()
+        );
 
         final Observation observation = specimenBatteryMapper.mapBatteryObservation(batteryParameters);
 
-        assertThat(observation.getId()).isEqualTo(OBSERVATION_ID);
-        assertThat(observation.getIdentifierFirstRep().getSystem()).contains(PRACTISE_CODE);
-        assertThat(observation.getEffectiveDateTimeType().getValueAsString()).isEqualTo(OBSERVATION_EFFECTIVE.getValueAsString());
-        assertThat(observation.getIssuedElement().getValueAsString()).isEqualTo(OBSERVATION_ISSUED.getValueAsString());
-        assertThat(observation.getSpecimen().hasReference()).isTrue();
-        assertThat(observation.getSpecimen().getReference()).contains(SPECIMEN_ID);
-        assertThat(observation.getStatus()).isEqualTo(ObservationStatus.FINAL);
-        assertThat(observation.getMeta().getProfile().get(0).getValue()).contains(META_PROFILE_SUFFIX);
-        assertThat(observation.getComment()).isEqualTo(EXPECTED_COMMENT);
-        assertThat(observation.getContext().hasReference()).isTrue();
-        assertThat(observation.getContext().getReference()).contains(ENCOUNTER_ID);
+        assertThat(observation.getIssuedElement().asStringValue())
+            .isEqualTo(parseToInstantType("20220302105070").asStringValue());
+    }
+
+    @Test
+    public void When_MappingObservation_Expect_ObservationCorrectlyMapped() {
```

Comment:
Would hate for it to be mapped incorrectly 😛 .

Reply:
yeah I was at a loss for a better name :(

Reply:
Thats the best I can come up with!